Add Missing Selector Annotation error (#431)
unicode-org/message-format-wg: spec/formatting.md

    > Example invalid messages resulting in a Missing Fallback Variant error:
    >
    > ```
    > match {$one :func}
    > when 1 {Value is one}
    > when 2 {Value is two}
    > ```
    >
    > ```
    > match {$one :func} {$two :func}
    > when 1 * {First is one}
    > when * 1 {Second is one}
    > ```

>   - **Missing Selector Annotation errors** occur when the message

Reviewer: Don't dig the formatting hole deeper. When you introduce the term, do so in the form:

```suggestion
A **_Missing Selection Annotation error_** is an error that occurs when the _message_
```

That is, the form is:

> A **_thing_** is {definition} with _terms_ italicized.

> A **_Missing Selection Annotation error_** is an error that occurs when the _message_

Author: ```suggestion
  - A **_Missing Selector Annotation error_** is an error that occurs when the _message_
```

    contains a _selector_ that does not have an _annotation_,

>     or contains a _variable_ that does not reference a _declaration_ with an _annotation_.

Author: ```suggestion
    or contains a _variable_ that does not directly or indirectly reference a _declaration_ with an _annotation_.
```

>     > Example invalid messages resulting in a Missing Selector Annotation error:

Reviewer: This says "Missing Selector Annotation error" but the definition says "Missing Selection Annotation error". Choose one.

Also, the name should be italicized here.

Author: ```suggestion
    > Example invalid messages resulting in a _Missing Selector Annotation error_:
```
